src/main/java/de/numcodex/sq2cql/model/structured_query/ValueSetCriterion.java

```diff
@@ -49,6 +49,23 @@ public static ValueSetCriterion of(TermCode concept, TermCode... selectedConcept
     }
 
     public Container<BooleanExpression> toCql(MappingContext mappingContext) {
+        var expr = fullExpr(mappingContext);
+        if (expr.isEmpty()) {
+            throw new TranslationException("Failed to expand the concept %s.".formatted(concept));
+        }
+        return expr;
+    }
+
+    /**
+     * Builds an OR-expression with an expression for each concept of the expansion of {@code termCode}.
+     */
+    private Container<BooleanExpression> fullExpr(MappingContext mappingContext) {
+        return mappingContext.expandConcept(concept)
+                .map(termCode -> expr(mappingContext, termCode))
+                .reduce(Container.empty(), Container.OR);
+    }
+
+    private Container<BooleanExpression> expr(MappingContext mappingContext, TermCode termCode) {
         //TODO: expand the termCode, look into ConceptCriterion#fullExpr
```

Comment:
@alexanderkiel  the TODO can be removed now?

Reply:
Good catch, will remove it.